Gems/AWSCore/Code/Include/Framework/AWSApiRequestJob.h

```diff
@@ -295,56 +268,86 @@ namespace AWSCore
         /// Called when request has completed successfully.
         virtual void OnSuccess()
         {
-            if (m_queueOnSuccess)
-            {
-                AZStd::function<void()> callbackHandler = [this]()
-                {
-                    if (m_onSuccess)
-                    {
-                        m_onSuccess(this);
-                    }
-                    delete this;
-                };
-                AZ::TickBus::QueueFunction(callbackHandler);
-            }
         }
 
         /// Called when the request fails.
         virtual void OnFailure()
         {
-            if (m_queueOnFailure)
-            {
-                AZStd::function<void()> callbackHandler = [this]()
-                {
-                    if (m_onFailure)
-                    {
-                        m_onFailure(this);
-                    }
-                    delete this;
-                };
-                AZ::TickBus::QueueFunction(callbackHandler);
-            }
         }
 
         /// Called when request can't process and still requires cleanup (Specifically for our derived class Function which does not use auto delete)
         virtual void DoCleanup()
         {
-            if (m_queueDelete)
-            {
-                AZStd::function<void()> callbackHandler = [this]()
-                {
-                    delete this;
-                };
-                AZ::TickBus::QueueFunction(callbackHandler);
-            }
         }
 
     public:
+        using OnSuccessFunction = AZStd::function<void(AwsApiRequestJob* job)>;
+        using OnFailureFunction = AZStd::function<void(AwsApiRequestJob* job)>;
+
         template<class Allocator = AZ::SystemAllocator>
         static AwsApiRequestJob* Create(OnSuccessFunction onSuccess, OnFailureFunction onFailure = OnFailureFunction{}, IConfig* config = GetDefaultConfig())
         {
-            return azcreate(AwsApiRequestJob, (onSuccess, onFailure, config), Allocator);
+            return azcreate(Function, (onSuccess, onFailure, config), Allocator);
```

Comment:
do you need to pull this function definition out after Function class definition? I remember Steve has mentioned it will fail

Reply:
I just moved the Create function implementation outside after the 'Function' class definition. Sounds like that's the solution that should work with GCC for sure.

Reply:
Verified that this compiles with gcc

Reply:
Thanks for the verification, Steve.